src/Execution/ModelNotFoundErrorHandler.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Nuwave\Lighthouse\Execution;
+
+use Closure;
+use GraphQL\Error\Error;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+/**
+ * Wrap model not found exceptions.
+ */
+class ModelNotFoundErrorHandler implements ErrorHandler
```

Comment:
I am not convinced that any `ModelNotFoundException` ever thrown is generally safe to show to users. What if it is not a model the user even queried, but rather just something happening in the background that the user is not supposed to know about? This handler captures anything that happens inside any query or mutation.

Reply:
> just removing the try catch inside the canDirective

How would that be better then the previous behaviour? I am wondering if we should even throw if no models are found at all - would it be so bad to continue with an empty result?

Recently, there has been a proposal to change the behaviour of `@can` in regards to not found models - see https://github.com/nuwave/lighthouse/issues/2225. Maybe we can work out an optimal solution there.